src/repositories/request-repository.ts

```diff
@@ -35,164 +31,215 @@ const countRetryMetrics = (requests: Request[], anchoringDeadline: Date): void =
-          )
+          return requests
+        },
+        {
+          isolationLevel: 'repeatable read',
         }
-
-        return requests
-      })
+      )
       .catch(async (err) => {
         if (err?.code === REPEATED_READ_SERIALIZATION_ERROR) {
           Metrics.count(METRIC_NAMES.DB_SERIALIZATION_ERROR, 1)
           await Utils.delay(100)
-          return this.findAndMarkAsProcessing()
+          return this.findAndMarkAsProcessing(options)
         }
 
         throw err
       })
   }
 
   /**
-   * Creates new client request
-   * @param cid: Client request CID
+   * Finds a request with the given CID if exists
+   * @param cid CID the request is for
+   * @param options
+   * @returns Promise for the associated request
    */
-  public async findByCid(cid: CID): Promise<Request> {
-    return await this.connection
-      .getRepository(Request)
-      .createQueryBuilder('request')
-      .where('request.cid = :cid', { cid: cid.toString() })
-      .getOne()
+  public async findByCid(cid: CID, options: Options = {}): Promise<Request> {
+    const { connection = this.connection } = options
+
+    return connection(TABLE_NAME).where({ cid: cid.toString() }).first()
   }
 
   /**
    * Gets all requests that were anchored over a month ago, and that are on streams that have had
    * no other requests in the last month.
+   * @param options
+   * @returns A promise that resolves to an array of request
    */
-  public async findRequestsToGarbageCollect(): Promise<Request[]> {
+  public async findRequestsToGarbageCollect(options: Options = {}): Promise<Request[]> {
+    const { connection = this.connection } = options
+
     const now: number = new Date().getTime()
     const deadlineDate = new Date(now - ANCHOR_DATA_RETENTION_WINDOW)
 
-    const expiredRequests = await this.connection
-      .getRepository(Request)
-      .createQueryBuilder('request')
-      .orderBy('request.updated_at', 'DESC')
-      .where('(request.status = :anchoredStatus1 OR request.status = :anchoredStatus2)', {
-        anchoredStatus1: RequestStatus.COMPLETED,
-        anchoredStatus2: RequestStatus.FAILED,
-      })
-      .andWhere('request.pinned = :pinned', { pinned: true })
-      .andWhere('request.updated_at < :deadlineDate', { deadlineDate: deadlineDate.toISOString() })
-
-    return expiredRequests
-      .andWhere((qb) => {
-        const recentRequestsStreamIds = qb
-          .subQuery()
-          .select('doc_id')
-          .from(Request, 'recent_request')
-          .orderBy('recent_request.updated_at', 'DESC')
-          .where('recent_request.updated_at >= :deadlineDate', {
-            deadlineDate: deadlineDate.toISOString(),
-          })
-          .getQuery()
-        return 'request.doc_id NOT IN ' + recentRequestsStreamIds
-      })
-      .getMany()
+    const recentlyUpdatedRequests = connection(TABLE_NAME)
```

Comment:
`requestsOnRecentlyUpdatedStreams`?
Something to make clear that we're not actually just skipping requests that were recently updated (which would already be handled by the `andWhere('updatedAt', '<', deadlineDate)`), but we also want to skip requests that haven't been updated recently directly, but where there are other requests that *were* updated recently on the same streamid